File: tests/csharp/tests-csharp-xunit-allure3-playwright/Helpers/ViewportHelper.cs
using Allure.Net.Commons.Attributes;
using Config;
using Microsoft.Playwright;

namespace Helpers;

public static class ViewportHelper
{
    [ThreadStatic]
    private static IPage? _page;

    public static void Bind(IPage page) => _page = page;

    public static void Unbind() => _page = null;

    public static IPage Page =>
        _page ?? throw new InvalidOperationException("ViewportHelper.Bind() was not called for this thread");

    [AllureStep("Reset viewport to default browser size")]
    public static void ResetViewport()
    {
        if (_page == null)
        {
            return;
        }

        var size = ParseBrowserSize(ConfigReader.TestConfig.BrowserSize);
        Pw.Run(_page.SetViewportSizeAsync(size.Width, size.Height));
    }

    public static void SetViewport(int width, int height) =>
        Pw.Run(Page.SetViewportSizeAsync(width, height));

    private static (int Width, int Height) ParseBrowserSize(string browserSize)
    {
        var parts = browserSize.Split('x');
        if (parts.Length != 2)
        {
            throw new InvalidOperationException("Invalid browserSize: " + browserSize);
        }

        return (int.Parse(parts[0].Trim()), int.Parse(parts[1].Trim()));
    }
}

File: tests/java/tests-java-junit5-api_request-playwright/src/test/java/helpers/ViewportHelper.java
package helpers;

import com.microsoft.playwright.Page;
import config.ConfigReader;
import io.qameta.allure.Step;

public final class ViewportHelper {

    private static final ThreadLocal<Page> PAGE = new ThreadLocal<>();

    private ViewportHelper() {
    }

    public static void bind(Page page) {
        PAGE.set(page);
    }

    public static void unbind() {
        PAGE.remove();
    }

    public static Page page() {
        var page = PAGE.get();
        if (page == null) {
            throw new IllegalStateException("ViewportHelper.bind() was not called for this thread");
        }
        return page;
    }

    @Step("Reset viewport to default browser size")
    public static void resetViewport() {
        var page = PAGE.get();
        if (page == null) {
            return;
        }
        var size = parseBrowserSize(ConfigReader.testConfig.browserSize());
        page.setViewportSize(size.width(), size.height());
    }

    public static void setViewport(int width, int height) {
        page().setViewportSize(width, height);
    }

    private static Viewport parseBrowserSize(String browserSize) {
        var parts = browserSize.split("x");
        if (parts.length != 2) {
            throw new IllegalStateException("Invalid browserSize: " + browserSize);
        }
        return new Viewport(Integer.parseInt(parts[0].trim()), Integer.parseInt(parts[1].trim()));
    }

    private record Viewport(int width, int height) {
    }
}
